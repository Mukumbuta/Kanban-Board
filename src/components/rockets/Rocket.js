/* eslint-disable react/prop-types */
import React from 'react';
import RocketCard from './RocketCard';

const Rocket = ({ rockets }) => (
  <div>
    {rockets.map((rocket) => (
      <RocketCard
        key={rocket.id}
        id={rocket.id}
        flickr_image={rocket.flickr_images[1]}
        description={rocket.description}
        rocket_name={rocket.rocket_name}
        reserved={rocket.reserved}
      />
    ))}

  </div>
);

export default Rocket;
