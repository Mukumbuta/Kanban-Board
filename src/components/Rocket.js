import React from 'react';
import PropTypes from 'prop-types';
import '../App.css';
import store from '../redux/configureStore';
import { reserveRocket } from '../redux/rockets/rockets';

import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import { Badge } from 'react-bootstrap';

const Rocket = ({ rockets }) => {
  
  return (
    <div>
      {rockets.map((rocket) => (
        <Card key={rocket.id} className="rocket-card m-4">
          <Card.Img style={{ width: '20rem' }} variant="card-img-top" src={rocket.flickr_images[1]} />
          <Card.Body>
            <Card.Title>{rocket.rocket_name}</Card.Title>
            <Card.Text>
             {rocket.reserved === true ? <Badge bg="success" className="m-1">Reserved</Badge> : "" }{rocket.description}
            </Card.Text>
            <Button
              type="button"
              variant={ rocket.reserved === true ? "outline-secondary" : "primary"} 
              onClick={() => {
                store.dispatch(reserveRocket(rocket.id));
              }}
            >
               {rocket.reserved === true ? "Cancel Reservations" : "Reserve Rocket"}
            </Button>
          </Card.Body>
        </Card>
      ))}

    </div>
  );
};

Rocket.propTypes = {
  rockets: PropTypes.arrayOf(
    PropTypes.shape({
      rocket_name: PropTypes.string,
    }),
  ).isRequired,
};

export default Rocket;
