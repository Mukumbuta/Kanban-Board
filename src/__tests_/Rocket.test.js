import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import TestRender from 'react-test-renderer';
import DisplayRocket from '../components/rockets/DisplayRockets';
import Rocket from '../components/rockets/DisplayRockets';

describe('Test DisplayRocket ', () => {
  it('Test DisplayRocket component ', () => {
    const component = TestRender.create(
      <Provider store={store}>
        <DisplayRocket />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Test Rocket ', () => {
  it('Test Rocket component ', () => {
    const component = TestRender.create(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});