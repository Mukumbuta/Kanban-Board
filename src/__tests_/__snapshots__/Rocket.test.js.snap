// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for DisplayRocket Component Matches the snapshot 1`] = `
<div>
  <div />
</div>
`;

exports[`snapshot test for RocketCard Component Matches the snapshot 1`] = `
<div
  className="rocket-card m-4 card"
>
  <img
    className="card-img-card-img-top"
    style={
      Object {
        "width": "20rem",
      }
    }
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    />
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
